### DISCUSSION TITLE
How do you change the icons of the ctk window?

### DISCUSSION THREAD

**Discussion by moyurun0321** [TomSchimansky/CustomTkinter#1295] [General]
I want to change this windows icon of the ctk window.
![image](https://user-images.githubusercontent.com/107347365/222059421-6344c58d-0de2-432c-90e7-34e60a471647.png)
and this window's icon just like any other app out there.
![image](https://user-images.githubusercontent.com/107347365/222059319-ce5a6b7c-a54a-4a4e-9be6-7eef0a304165.png)

Thank you

**Comment by Akascape** [comment]
```python
from PIL import ImageTk

root = customtkinter.CTk()
root.wm_iconbitmap()
icopath = ImageTk.PhotoImage(file="icon.png")
root.iconphoto(False, icopath)
```

**Reply by LucEast** [reply]
Doesn't work on mac tho. 
<img width="548" alt="grafik" src="https://user-images.githubusercontent.com/47122750/224403863-3b5e8698-6528-418c-adf5-8cf2ae4af215.png">

**Reply by Akascape** [reply]
@LucEast 
Try this:
```python
root.wm_iconbitmap("your_icon.ico")
```

**Reply by LucEast** [reply]
Same problem.

**Reply by Akascape** [reply]
@LucEast Make sure you are using a proper png image with `iconphoto` method.
In my system, the first method works:

![Screenshot](https://user-images.githubusercontent.com/89206401/224480497-d19d4892-cc08-4ec8-bbbc-246737f7afb2.jpg)

**Reply by LucEast** [reply]
I got it working with the following:
```python
self.iconbitmap()
icopath = ImageTk.PhotoImage(file=resource("images\krz_school.png"), master=self)
self.iconphoto(True, icopath)
```
But I ran into another problem.
Even though I set `self.iconphoto()` to `True` it isn't applied to Toplevel Windows.

**Reply by Akascape** [reply]
@LucEast That is a bug in the ctktoplevel. I have raised an issue.

**Comment by moyurun0321** [comment]
Thank you!!

**Comment by moyurun0321** [comment]
Sorry, I got this error

   Icopath =
ImageTk.PhotoImage(file=resource("C:\Users\moyur\Desktop\Untitled
design.png"))

               ^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in
position 2-3: truncated \UXXXXXXXX escape

**Comment by moyurun0321** [comment]
Sorry, I solved it myself!

**Comment by greeenboi** [comment]
I have a similar issue
mind helping me?
```
import tkinter
from tkinter.ttk import *
from PIL import ImageTk
...
if __name__ == "__main__":
    app = App()    
    icopath = ImageTk.PhotoImage(file="snowflake.png")
    app.iconphoto(False, icopath)
    app.mainloop()
```

The error is 
```
File "D:\practice python\main.py", line 174, in <module>
  app.iconphoto(False, icopath)

File "D:\anaconda\lib\tkinter\__init__.py", line 2127, in wm_iconphoto
  self.tk.call('wm', 'iconphoto', self._w, *args)

TclError: can't use "pyimage7" as iconphoto: not a photo image
```


What can i do to fix this issue?

**Reply by Akascape** [reply]
You can add the master parameter:
```python
icopath = ImageTk.PhotoImage(file="snowflake.png", master=app)
```

**Reply by greeenboi** [reply]
ok It doesn't return an error but the logo hasn't changed.
Its still the default.

**Reply by greeenboi** [reply]
ok never mind thanks for your help. 
It was a path specifier issue; forgot to add ./
